src/ansys/geometry/core/designer/design.py

```diff
@@ -351,14 +364,18 @@ def __export_and_download(self, format: DesignFileFormat) -> bytes:
             DesignFileFormat.STRIDE,
         ]:
             try:
-                response = self._grpc_client.services.designs.download_export(format=format)
+                response = self._grpc_client.services.designs.download_export(
+                    format=format, write_body_facets=write_body_facets
+                )
             except Exception:
                 self._grpc_client.log.warning(
                     f"Failed to download the file in {format} format."
                     " Attempting to stream download."
                 )
                 # Attempt to download the file via streaming
-                response = self._grpc_client.services.designs.stream_download_export(format=format)
+                response = self._grpc_client.services.designs.stream_download_export(
+                    format=format, write_body_facets=write_body_facets
```

Comment:
Agreed.. the write_body_facets, in case it is True, should be verified against the backend version. If True and backend_version < 26.1 then we need to throw a warning and inform that this option will not be considered since the feature wasn't available prior to 26.1

Reply:
This comment applies to the three methods. Maybe you want to come up with a common method and reuse it. Something like ``check_write_body_facets_input(write_body_facets, backend_version)``.

Reply:
I would do something like this in fact:

```py
        # HACK: we should inform the user that NURBS curve sketches are not supported
        # prior to 26R1... and if passed, raise an error
        _nurbs_curves_compatibility(kwargs["backend_version"], request.geometries)
 ```
 
 This is already being done inside the ``_grpc/_services/v0/bodies.py`` file. Could you implement this @jonahrb ?